internal/controller/scheduler/scheduler.go

```diff
@@ -1272,7 +1273,14 @@ func (w *worker) failJob(ctx context.Context, inputs buildInputs, message string
 	}
 
 	opts := w.cfg.AgentConfig.ControllerOptions()
-	if err := acquireAndFail(ctx, w.logger, agentToken, w.cfg.JobPrefix, inputs.uuid, inputs.agentQueryRules, message, opts...); err != nil {
+	failureInfo := FailureInfo{
+		Message: message,
+		// In scheduler worker, often these failure are technically users error in YAML.
+		// Using agent refuse seems to be reasonable as that represent that agent won't execute these yaml.
+		// Deserve more discussion though..
+		Reason: agent.SignalReasonAgentRefused,
```

Comment:
@DrJosh9000 I think this change is debatable, subject to our confidence level. Just wanting to highlight this in case in you have opinion here

Reply:
Hmm. I think the cases here where `failJob` is called are all "pre-agent", so using "agent refused" feels wrong for that reason. We also can't rule out the Kubernetes cluster having a bad day and rejecting a valid job. "Stack error" seems like the best choice for now.

Maybe the agent change should have included a "stack rejected" reason, as well as "stack error"?

Reply:
Ah true, the controller isn't an agent. I changed it to stack error but with a bit comments explaining the caveats. 🙏🏿